src/nl/rubensten/texifyidea/settings/TexifySettings.kt

```diff
@@ -24,9 +25,50 @@ class TexifySettings : PersistentStateComponent<TexifySettings> {
     var automaticUpDownBracket = true
     var automaticItemInItemize = true
 
+    /**
+     * internal list which stores the commands data
+     */
+    private val _labelCommands: HashMap<String, LabelingCommandInformation> =
```

Comment:
Please don't use member prefixing. I find it weird that you need to use a backing field anyway.. Maybe find a way to join these in some form.

```kotlin
    /**
     * Internal list that stores the label defining commands.
     */
    private val labelCommandsOrABetterName: HashMap<String, LabelingCommandInformation> = hashMapOf(
            "\\label" to LabelingCommandInformation("\\label", 1, true)
    )
```

Reply:
The backing field is needed to initialize the Map in a not empty state. With the other property I could verify that every key of the map has a leading backslash.

Reply:
Then you probably need a function that verifies this instead of a field. The member-prefixing still holds: don't do it :)

Maybe also take a look at [`Lazy`](https://kotlinlang.org/docs/reference/delegated-properties.html#lazy)?

Reply:
Could you explain why you wont use backing fields? I, personaly, think this is one of the best parts about kotlin, because I only have one field containing data, but could represent it in several to other functions.
My prefered way would be to write the `get()` directly under the field, but you can't initialize a property without a backing field. The reason why I don't use a function was that I wanted to do it in a more kotlin idiomatic style without a getter function.

Reply:
@Stainzl @RubenSchellekens I think backing fields are sometimes necessary in Kotlin to provide some modification to the underlying data structure. However, in this particular case, I find it a bit odd. Especially since the getter performs a map on the keys on every access. This is an (undocumented) and unconventional side effect of the getter; the caller does not expect this calculation every time.

For me, this raises the question why the ```addLeadingSlash``` logic is placed here. I think this is not the responsibility of the ```TexifySettings``` class, but rather of the classes that directly manipulate ```labelCommands```.

Therefore, you could simply use ```val labelCommands = hashMapOf(...)``` directly and expose the mutable map. This is consistent with the other class members as everything is publicly modifiable already. Note in the docstring keys are required to have a leading slash. I think this is sufficient.

Side note: if you really really want to use a backing field here (which I think you shouldn't), do something like this (reducing the computations in the getter):
```kotlin
private val _labelCommands: HashMap<String, LabelingCommandInformation> = hashMapOf(...)
val labelCommands: Map<String, LabelingCommandInformation> get() =  _labelCommands

fun addLabelCommand(key: String, info: LabelingCommandInformation) {
    _labelCommands[key.addLeadingSlash()] = info
}
```

Reply:
Okay, with the use of a external state class the backing field couold be removed.